apiserver crashes in mutating.callAttrMutatingHook

**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

apiserver crashes, and continue to crash, with the following error message

```
E0815 02:17:18.379437     320 runtime.go:66] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:509
/usr/local/go/src/runtime/panic.go:491
/usr/local/go/src/runtime/panic.go:63
/usr/local/go/src/runtime/signal_unix.go:367
/workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/admission.go:296
/workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/admission.go:244
/workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/metrics/metrics.go:85
<autogenerated>:1
/workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/chain.go:35
<autogenerated>:1
/workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/metrics/metrics.go:85
<autogenerated>:1
/workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/patch.go:98
/workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/patch.go:354
/workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/registry/rest/update.go:184
/workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/registry/generic/registry/store.go:529
/workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/storage/etcd3/store.go:734
/workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/storage/etcd3/store.go:305
/workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/storage/cacher.go:553
/workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/registry/generic/registry/store.go:527
/workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/patch.go:359
/workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/rest.go:182
/usr/local/go/src/runtime/asm_amd64.s:2337
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x14aec8f]

goroutine 6165 [running]:
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0xc449137f78, 0x1, 0x1)
        /workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x111
panic(0x33bd4c0, 0x9059170)
        /usr/local/go/src/runtime/panic.go:491 +0x283
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating.(*MutatingWebhook).callAttrMutatingHook(0xc420302c80, 0x8ecad20, 0xc420014050, 0xc45e72ef00, 0x8ed6f40, 0xc44aee9930, 0x8eae8e0, 0xc445dd1c00, 0x8eae8e0, 0xc450b58000, ...)
        /workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/admission.go:296 +0x2ff
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating.(*MutatingWebhook).Admit(0xc420302c80, 0x8ed6f40, 0xc44aee9930, 0x7f46c856e280, 0xc420302c80)
        /workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/admission.go:244 +0x405
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/metrics.pluginHandlerWithMetrics.Admit(0x8e9e220, 0xc420302c80, 0xc4201935e0, 0xc4201935f0, 0x1, 0x1, 0x8ed6f40, 0xc44aee9930, 0x1, 0x7f46c856e130)
        /workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/metrics/metrics.go:85 +0xca
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/metrics.(*pluginHandlerWithMetrics).Admit(0xc420f2b560, 0x8ed6f40, 0xc44aee9930, 0x7f46c856e130, 0xc420f2b560)
        <autogenerated>:1 +0x5f
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission.chainAdmissionHandler.Admit(0xc4200c4800, 0xe, 0x10, 0x8ed6f40, 0xc44aee9930, 0x1114d1b2, 0x1c281c373)
        /workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/chain.go:35 +0xf5
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission.(*chainAdmissionHandler).Admit(0xc420f38b80, 0x8ed6f40, 0xc44aee9930, 0x7f46c856e190, 0xc420f38b80)
        <autogenerated>:1 +0x62
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/metrics.pluginHandlerWithMetrics.Admit(0x8ea4c60, 0xc420f38b80, 0xc4202a20f0, 0x0, 0x0, 0x0, 0x8ed6f40, 0xc44aee9930, 0x1, 0xc44aee9930)
        /workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/metrics/metrics.go:85 +0xca
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/metrics.(*pluginHandlerWithMetrics).Admit(0xc420f43020, 0x8ed6f40, 0xc44aee9930, 0x7f46c856e101, 0xc420f43020)
        <autogenerated>:1 +0x5f
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers.PatchResource.func1.1(0x8ea8260, 0xc42d465500, 0x8ea8260, 0xc42d464e00, 0xc44d3f3345, 0xc4547c7638)
        /workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/patch.go:98 +0x391
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers.patchResource.func2(0x7f46c8897ac0, 0xc440c2c5d0, 0x8ea8260, 0xc42d465500, 0x8ea8260, 0xc42d464e00, 0x8ea8260, 0xc42d465500, 0x0, 0x0)
        /workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/patch.go:354 +0x4d
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/registry/rest.(*defaultUpdatedObjectInfo).UpdatedObject(0xc440c2c870, 0x7f46c8897ac0, 0xc440c2c5d0, 0x8ea8260, 0xc42d464e00, 0xc4547c76a0, 0x46d169, 0x0, 0x0)
        /workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/registry/rest/update.go:184 +0xbd
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/registry/generic/registry.(*Store).Update.func1(0x8ea8260, 0xc42d464e00, 0x0, 0x2d266a3c, 0x0, 0x0, 0x2d266a3c, 0xc42a869340, 0xc42a868fa8)
        /workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/registry/generic/registry/store.go:529 +0x107
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/storage/etcd3.(*store).updateState(0xc4200613b0, 0xc42a869340, 0xc42e64b0e0, 0x8ea8260, 0xc42d464e00, 0x0, 0x0, 0x0)
        /workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/storage/etcd3/store.go:734 +0x5b
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/storage/etcd3.(*store).GuaranteedUpdate(0xc4200613b0, 0x8ecada0, 0xc440c2c5d0, 0xc450aa9ad0, 0x23, 0x8ea8260, 0xc42d464a80, 0x1, 0xc421344580, 0xc42e64b0e0, ...)
        /workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/storage/etcd3/store.go:305 +0x4c4
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/storage.(*Cacher).GuaranteedUpdate(0xc42135c000, 0x8ecada0, 0xc440c2c5d0, 0xc450aa9ad0, 0x23, 0x8ea8260, 0xc42d464a80, 0x1, 0xc421344580, 0xc42e64b0e0, ...)
        /workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/storage/cacher.go:553 +0x2e8
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/registry/generic/registry.(*Store).Update(0xc420223e00, 0x7f46c8897ac0, 0xc440c2c5d0, 0xc45d478897, 0x14, 0x8eb10e0, 0xc440c2c870, 0xc440c2c630, 0xc440c2c840, 0x0, ...)
        /workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/registry/generic/registry/store.go:527 +0x442
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers.patchResource.func3(0x18, 0x3c22c38, 0xc45d313778, 0x1)
        /workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/patch.go:359 +0xe3
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers.finishRequest.func1(0xc4415e4960, 0xc4415e4840, 0xc4415e4900, 0xc4415e48a0)
        /workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/rest.go:182 +0x8c
created by k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers.finishRequest
        /workspace/anago-v1.9.9-beta.0.51+57729ea3d9a1b7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/rest.go:175 +0xd4
```

**What you expected to happen**:

Not crash?   😉 

**How to reproduce it (as minimally and precisely as possible)**:

Not sure, working on reproducing it.   It happened while I was working on a patched version of vpa-admission-controller.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v1.9.9
- Cloud provider or hardware configuration: bare metal
- OS (e.g. from /etc/os-release): CentOS 7
- Kernel (e.g. `uname -a`): 4.14.51
- Install tools: custom
- Others:

--- a/staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/admission.go
+++ b/staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/admission.go
@@ -293,6 +293,10 @@ func (a *MutatingWebhook) callAttrMutatingHook(ctx context.Context, h *v1beta1.W
 		return &webhookerrors.ErrCallingWebhook{WebhookName: h.Name, Reason: err}
 	}
 
+	if response.Response == nil {
+		return &webhookerrors.ErrCallingWebhook{WebhookName: h.Name, Reason: fmt.Errorf("webhook response was absent")}
+	}
+
 	if !response.Response.Allowed {
 		return webhookerrors.ToStatusErr(h.Name, response.Response.Result)
 	}

